google-beta/resource_bigquery_table.go

```diff
@@ -62,6 +63,182 @@ func resourceBigQueryTable() *schema.Resource {
 				Computed: true,
 			},
 
+			// ExternalDataConfiguration [Optional] Describes the data format,
+			// location, and other properties of a table stored outside of BigQuery.
+			// By defining these properties, the data source can then be queried as
+			// if it were a standard BigQuery table.
+			"external_data_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Autodetect : [Required] If true, let BigQuery try to autodetect the
+						// schema and format of the table.
+						"autodetect": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						// BigtableOptions: [Optional] Additional options if sourceFormat is set to BIGTABLE.
+						"bigtable_options": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									// ColumnFamilies: [Optional] List of column families to expose in the
+									// table schema along with their types.
+									"column_families": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										Computed:     true,
+										ValidateFunc: validation.ValidateJsonString,
+										StateFunc: func(v interface{}) string {
+											json, _ := structure.NormalizeJsonString(v)
+											return json
+										},
+									},
+									// IgnoreUnspecifiedColumnFamilies: [Optional] If field is true, then
+									// the column families that are not specified in columnFamilies list are
+									// not exposed in the table schema.
+									"ignore_unspecified_column_families": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									// ReadRowkeyAsString: [Optional] If field is true, then the rowkey
+									// column families will be read and converted to string.
+									"read_rowkey_as_string": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+								},
+							},
+						},
+						// Compression: [Optional] The compression type of the data source.
+						"compression": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"NONE", "GZIP"}, false),
+						},
+						// CsvOptions: [Optional] Additional properties to set if
+						// sourceFormat is set to CSV.
+						"csv_options": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									// AllowJaggedRows: [Optional] Indicates if BigQuery should
+									// accept rows that are missing trailing optional columns.
+									"allow_jagged_rows": {
```

Comment:
Can we explicitly default to `false` as per the docs safely?

Reply:
Added this in https://github.com/GoogleCloudPlatform/magic-modules/pull/1466/commits/1e5032ff09f690276e386a31909ab2f697b95f18